src/mumble/GlobalShortcut_win.cpp

```diff
@@ -395,11 +394,35 @@ void GlobalShortcutWin::run() {
 		qWarning("GlobalShortcutWindows: RegisterRawInputDevices() failed with error %u!", GetLastError());
 	}
 
-	QTimer timer;
-	connect(&timer, &QTimer::timeout, this, &GlobalShortcutWin::timeTicked);
-	timer.start(20);
+	std::mutex mutex;
+	std::unique_lock< decltype(mutex) > lock(mutex);
+
+	do {
+		m_condVar.wait(lock);
+
+		if (bNeedRemap) {
+			remap();
+		}
+
+		while (const auto item = m_msgQueue.front()) {
```

Comment:
Invalid use of `auto`

Reply:
This is the queue's item pointer, I think it's fine.

Reply:
I don't agree. We have a local variable `item` whose type is completely inaccessible by simply looking at this piece of code.